class-toc-plus.php

```diff
@@ -325,23 +325,45 @@ public function shortcode_sitemap( $atts ) {
 
 
 		public function shortcode_sitemap_pages( $attributes ) {
+			global $post;
+			$args_child_of = 0;
 			$atts = shortcode_atts(
 				[
 					'heading'      => $this->options['sitemap_heading_type'],
 					'label'        => $this->options['sitemap_pages'],
 					'no_label'     => false,
 					'exclude'      => '',
 					'exclude_tree' => '',
+					'child_of'     => $this->options['child_of']
 				],
 				$attributes
 			);
 
+			// -- debugging
+			//echo 'debug: child_of:   ' . $child_of . ' ' . "\n";    // true
+			//echo 'debug: current ID: ' . $post->ID . ' end ' . "\n";  // 2311
+
+
 			$atts['heading'] = intval( $atts['heading'] );  // make sure it's an integer
 
 			if ( $atts['heading'] < 1 || $atts['heading'] > 6 ) {  // h1 to h6 are valid
 				$atts['heading'] = $this->options['sitemap_heading_type'];
 			}
 
+
+
+			// -- [ sitemap_pages child_of="current" ]
+			if ( $atts['child_of'] == "current" ) {
+				$args_child_of = $post->ID;
+			}
+			else if ( is_numeric($child_of) ) {
+				// -- [ sitemap_pages child_of="1234" ]
+				$args_child_of = $child_of;
```

Comment:
`$child_of` shouldn't exist as we haven't used `extract()` from the shortcode attributes so this reference is incorrect? Use `$atts['child_of']` instead.